Core/Rewriters/NoMoqRewriter.cs

```diff
@@ -0,0 +1,188 @@
+      symbols.AddRange (GetAllRhinoMocksLocalDeclarationSymbols (node, rhinoMocksMockRepositoryCompilationSymbol));
+
+      return symbols;
+    }
+
+    private IEnumerable<ISymbol> GetAllRhinoMocksLocalDeclarationSymbols (SyntaxNode node, INamedTypeSymbol rhinoMocksMockRepositoryCompilationSymbol)
+    {
+      return node
+          .SyntaxTree.GetRoot()
+          .DescendantNodes()
+          .Where (s => s.IsKind (SyntaxKind.LocalDeclarationStatement))
+          .Select (s => (LocalDeclarationStatementSyntax) s)
+          .Select (s => s.Declaration)
+          .Where (s => rhinoMocksMockRepositoryCompilationSymbol.Equals (Model.GetSymbolInfo (s.Type).Symbol, SymbolEqualityComparer.Default))
+          .SelectMany (s => s.Variables)
+          .Select (s => Model!.GetDeclaredSymbol (s) ?? throw new InvalidOperationException ("Cannot resolve Symbol from VariableDeclarator"));
+    }
+
+    private IEnumerable<ISymbol> GetAllRhinoMocksFieldDeclarationSymbols (SyntaxNode node, INamedTypeSymbol rhinoMocksMockRepositoryCompilationSymbol)
+    {
+      return node
+          .SyntaxTree.GetRoot()
+          .DescendantNodes()
+          .Where (s => s.IsKind (SyntaxKind.FieldDeclaration))
+          .Select (s => (FieldDeclarationSyntax) s)
+          .Where (s => rhinoMocksMockRepositoryCompilationSymbol.Equals (Model.GetSymbolInfo (s.Declaration.Type).Symbol, SymbolEqualityComparer.Default))
+          .SelectMany (s => s.Declaration.Variables)
+          .Select (s => Model!.GetDeclaredSymbol (s) ?? throw new InvalidOperationException ("Cannot resolve Symbol from VariableDeclarator"));
+    }
+
+    private IEnumerable<SyntaxNode> GetAllNodesToBeReplaced (
+        MethodDeclarationSyntax node,
+        INamedTypeSymbol rhinoMocksMockRepositoryCompilationSymbol,
+        INamedTypeSymbol rhinoMocksExtensionsCompilationSymbol)
+    {
+      var allNodesToBeReplaced = new List<SyntaxNode>();
+      allNodesToBeReplaced.AddRange (GetReplayAndReplayAllExpressionStatements (node, rhinoMocksExtensionsCompilationSymbol, rhinoMocksMockRepositoryCompilationSymbol));
+      allNodesToBeReplaced.AddRange (GetAllRhinoMocksLocalDeclarationStatements (node, rhinoMocksMockRepositoryCompilationSymbol));
+      allNodesToBeReplaced.AddRange (GetAllRhinoMocksAssignmentExpressions (node));
+
+      return allNodesToBeReplaced;
+    }
+
+    private IEnumerable<ExpressionStatementSyntax> GetAllRhinoMocksAssignmentExpressions (MethodDeclarationSyntax node)
+    {
+      return node.DescendantNodes()
+          .Where (s => s.IsKind (SyntaxKind.ExpressionStatement))
+          .Select (s => (ExpressionStatementSyntax) s)
+          .Where (
+              s => s.Expression is AssignmentExpressionSyntax assignmentExpression
+                   && _mockRepositorySymbols!.Contains (Model!.GetSymbolInfo (assignmentExpression.Left).Symbol, SymbolEqualityComparer.Default));
```

Comment:
Is having `_mockRepositorySymbols` really easier than just checking the type of the localdecl here? There might be a benefit I'm not seeing.